### ISSUE TITLE
data types update - DB sources

### ISSUE THREAD

**Issue by edgao** [airbytehq/airbyte#17636]
Once https://github.com/airbytehq/airbyte/pull/17486 is approved and merged, we can start upgrading connectors to the new protocol. We shouldn't publish these connectors until https://github.com/airbytehq/airbyte/issues/17639 is released, though.

These changes need to happen in each source connector:
1. `discover` should return catalogs using the new well-known type definitions, rather than basic `{type: x, airbyte_type: y}` declarations. See https://github.com/airbytehq/airbyte/pull/17489 for an example of the new catalog
2. `read` should wrap numeric values (int/float/etc) inside strings
3. double-check that sources produce timestamps in the exact format described in the spec (specifically: `T` separator but not space; timezone as either `Z`, `+HH:MM`, or `-HH:MM` but not `+/-HH`)

For bonus points:
* Add special handling for binary-typed columns, which can be discovered as the new BinaryData type rather than plain strings
* Add support for infinity/nan in source-postgres (and presumably other sources) https://github.com/airbytehq/airbyte/issues/8902

**Comment by edgao** [comment]
@edgao call out jsonschematype class

**Comment by edgao** [comment]
@akashkulk when you add the infinity/nan and binary data handling, can you also update the postgres docs? (https://docs.airbyte.com/integrations/sources/postgres/#data-type-mapping)

cc @Amruta-Ranade

**Comment by akashkulk** [comment]
This is no longer relevant, closing
